src/contexts/TokenContext.tsx

```diff
@@ -25,8 +25,8 @@ export const TokenContext = createContext<TokenContextIF>({} as TokenContextIF);
 
 export const TokenContextProvider = (props: { children: React.ReactNode }) => {
     const { chainData } = useContext(CrocEnvContext);
-    const { tokenBalances } = useContext(UserDataContext);
-
+    // TODO: possible option to merge TokenBalanceContext with TokenContext
```

Comment:
To me TokenBalanceContext and TokenContext appear to be related concepts, maybe we can merge them.